OrbitGl/ManualInstrumentationManager.cpp

```diff
@@ -3,3 +3,54 @@
 // found in the LICENSE file.
 
 #include "ManualInstrumentationManager.h"
+
+using orbit_client_protos::TimerInfo;
+
+ManualInstrumentationManager::ManualInstrumentationManager(TimerInfoCallback callback) {
+  timer_info_callback_ = callback;
+}
+
+orbit_api::Event ManualInstrumentationManager::ApiEventFromTimerInfo(
+    const orbit_client_protos::TimerInfo& timer_info) {
+  // On x64 Linux, 6 registers are used for integer argument passing.
+  // Manual instrumentation uses those registers to encode orbit_api::Event
+  // objects.
+  constexpr size_t kNumIntegerRegisers = 6;
+  CHECK(timer_info.registers_size() == kNumIntegerRegisers);
+  uint64_t arg_0 = timer_info.registers(0);
+  uint64_t arg_1 = timer_info.registers(1);
+  uint64_t arg_2 = timer_info.registers(2);
+  uint64_t arg_3 = timer_info.registers(3);
+  uint64_t arg_4 = timer_info.registers(4);
+  uint64_t arg_5 = timer_info.registers(5);
+  orbit_api::EncodedEvent encoded_event(arg_0, arg_1, arg_2, arg_3, arg_4, arg_5);
+  return encoded_event.event;
+}
+
+void ManualInstrumentationManager::ProcessAsyncTimer(
+    const orbit_client_protos::TimerInfo& timer_info) {
+  orbit_api::Event event = ApiEventFromTimerInfo(timer_info);
+  if (event.type == orbit_api::kScopeStartAsync) {
+    async_timer_info_start_by_id_[event.id] = timer_info;
```

Comment:
This allows to use the same dummy id in the case where the async events are known to be non-overlapping. This will be refined in a PR that will make the event processing more robust (allowing for start and end events to be received out of order).